tests/nativeapp/test_event_sharing.py

```diff
@@ -0,0 +1,1086 @@
+        is_upgrade=is_upgrade,
+    )
+    mock_conn.return_value = MockConnectionCtx()
+    mock_diff_result = DiffResult()
+    _setup_project(
+        manifest_contents=manifest_contents,
+        authorize_event_sharing=authorize_event_sharing,
+    )
+    assert not mock_diff_result.has_changes()
+    mock_console = MagicMock()
+
+    _create_or_upgrade_app(
+        policy=MagicMock(),
+        install_method=install_method,
+        console=mock_console,
+    )
+
+    assert mock_execute_query.mock_calls == mock_execute_query_expected
+    mock_console.warning.assert_not_called()
+
+
+@mock.patch(APP_ENTITY_GET_EXISTING_APP_INFO, return_value=None)
+@mock.patch(SQL_EXECUTOR_EXECUTE)
+@mock_connection()
+@mock.patch(
+    GET_UI_PARAMETERS,
+    return_value={
+        UIParameter.EVENT_SHARING_V2: "true",
+        UIParameter.ENFORCE_MANDATORY_FILTERS: "false",
+    },
+)
+@pytest.mark.parametrize(
+    "manifest_contents",
+    [test_manifest_with_mandatory_events],
+)
+@pytest.mark.parametrize("authorize_event_sharing", [True, False, None])
+@pytest.mark.parametrize(
+    "install_method",
+    [
+        SameAccountInstallMethod.unversioned_dev(),
+        SameAccountInstallMethod.release_directive(),
+    ],
+)
+@pytest.mark.parametrize("is_upgrade", [False, True])
+def test_event_sharing_enabled_with_mandatory_events_and_explicit_authorization_then_flag_respected_with_potential_warning(
+    mock_param,
+    mock_conn,
+    mock_execute_query,
+    mock_get_existing_app_info,
+    manifest_contents,
+    authorize_event_sharing,
+    install_method,
+    is_upgrade,
+    temp_dir,
+    mock_cursor,
+):
+    mock_execute_query_expected = _setup_mocks_for_app(
+        mock_execute_query,
+        mock_cursor,
+        mock_get_existing_app_info,
+        is_prod=not install_method.is_dev_mode,
+        authorize_telemetry_flag=authorize_event_sharing,
+        is_upgrade=is_upgrade,
+    )
+    mock_conn.return_value = MockConnectionCtx()
+    mock_diff_result = DiffResult()
+    _setup_project(
+        manifest_contents=manifest_contents,
+        authorize_event_sharing=authorize_event_sharing,
+    )
+    assert not mock_diff_result.has_changes()
+    mock_console = MagicMock()
+
+    _create_or_upgrade_app(
+        policy=MagicMock(),
+        install_method=install_method,
+        console=mock_console,
+    )
+
+    assert mock_execute_query.mock_calls == mock_execute_query_expected
+    # Warn if the flag is not set, but there are mandatory events
+    if authorize_event_sharing:
+        mock_console.warning.assert_not_called()
+    else:
+        mock_console.warning.assert_has_calls(
+            [
+                mock.call(
+                    "WARNING: Mandatory events are present in the manifest file, but event sharing is not authorized in the application telemetry field. This will soon be required to set in order to deploy applications."
+                )
+            ]
+        )
```

Comment:
```suggestion
        mock_console.warning.assert_called_with(
            "WARNING: Mandatory events are present in the manifest file, but event sharing is not authorized in the application telemetry field. This will soon be required to set in order to deploy applications."
        )
```